doc/nrf/ug_bt_mesh_node_removal.rst

```diff
@@ -0,0 +1,190 @@
+
+The procedure of updating a network (application) key is described below.
+
+Uploading a new key
+===================
+
+Upload a new network key (or application key) to the nodes by sending the Config NetKey (AppKey) Update message to every node including the node with the Configuration Client model.
+
+The snippet below demonstrates how to do this using |NCS| API:
+
+.. code-block:: c
+
+    static uint8_t send_net_key_update(struct bt_mesh_cdb_node *node, void *user_data)
+    {
+        struct bt_mesh_cdb_subnet *subnet = user_data;
+        uint8_t status = 0;
+        int err;
+
+        err = bt_mesh_cfg_cli_net_key_update(node->net_idx, node->addr, subnet->net_idx,
+                                             subnet->keys[1].net_key, &status);
+        if (err || status) {
+            printk("Unable to update NetKey on %#x (err %d, status %u)\n", node->addr, err,
+                   status);
+        } else {
+            printk("Updated NetKey with index %u on %#x\n", subnet->net_idx, node->addr);
+        }
+
+        return BT_MESH_CDB_ITER_CONTINUE;
+    }
+
+    static int update_net_key(uint16_t net_idx, uint8_t *new_net_key)
+    {
+        struct bt_mesh_cdb_subnet *subnet;
+
+        subnet = bt_mesh_cdb_subnet_get(net_idx);
+        if (!subnet) {
+            printk("Unable to get subnet\n");
+            return -EINVAL;
+        }
+
+        /* Store the new network key in CDB. */
+        memcpy(subnet->keys[1].net_key, new_net_key, 16);
+        bt_mesh_cdb_subnet_store(subnet);
+
+        /* Send the new network key to each node. */
+        bt_mesh_cdb_node_foreach(send_net_key_update, subnet);
+        return 0;
+    }
+
+    ...
+
+    uint8_t new_net_key[16] = { ... };
+    update_net_key(BT_MESH_NET_PRIMARY, new_net_key);
+
+Switching to the new key
+========================
+
+After uploading the new key, command the nodes to use it when sending messages. To do that, change the Key Refresh Phase to 2 by sending the Config Key Refresh Phase Set message with Transition field set to 0x2.
+
+Send the message to at least one node. The rest nodes will switch the Key Refresh Phase through Secure Network Beacon. But, in this case, it may take a while for all nodes to eventually switch the phase since the time between 2 consecutive Secure Network Beacons is ~10 seconds. You can speed this process up by sending the Config Key Refresh Phase Set message to all nodes that needs to be updated.
+
+After switching the Key Refresh Phase to 2, a node will decrypt messages using both, old and new keys, but encrypt only using the new key. Therefore, when updating a network key, make sure to update the phase first on the node running the Configuration Client model so that it is able to decrypt the Config Key Refresh Phase Status message from other nodes when they switch to the new network key.
+
+The snippet below demonstrates how to set the Key Refresh Phase on all nodes using |NCS| API:
```

Comment:
```suggestion
Send the message to at least one node.
The rest of the nodes will switch the Key Refresh Phase through Secure Network Beacon.
But in this case, it may take a while for all nodes to eventually switch the phase since the time between 2 consecutive Secure Network Beacons is ~10 seconds.
You can speed up this process by sending the Config Key Refresh Phase Set message to all nodes that need to be updated.

After switching the Key Refresh Phase to 2, a node will decrypt messages using both old and new keys, but encrypt only using the new key.
Therefore, when updating a network key, make sure to update the phase on the node running the Configuration Client model first, so it is able to decrypt the Config Key Refresh Phase Status message from other nodes when they switch to the new network key.

The snippet below demonstrates how to set the Key Refresh Phase on all nodes using the |NCS| API:
```